conans/client/remote_manager.py

```diff
@@ -161,10 +204,9 @@ def addfile(name, file_info, tar):
 def uncompress_files(files, folder, name):
     try:
         for file_name, content in files:
-            if os.path.basename(file_name) != name:
-                save(os.path.join(folder, file_name), content)
-            else:
-                #  Unzip the file
+            save(os.path.join(folder, file_name), content)
```

Comment:
¿?¿? The tgz is not kept for saving a few seconds in the check, is kept to upload it directly without creating the tgz again.